lib/parsers/asm-parser-mlir.ts

```diff
@@ -46,11 +46,11 @@ export class MlirAsmParser extends AsmParser {
         this.locDefUnknownRegex = /^#(\w+)\s*=\s*loc\(unknown\)/;
 
         // Match location references like loc(#loc1)
-        this.locRefRegex = /loc\(#(\w+)\)/;
+        this.locRefRegex = / loc\(#(\w+)\)/;
```

Comment:
Is this strictly "exactly a space" or should it be either `\b` (for any word break) or `\w` (for any whitespace) ?

Reply:
From what I observe, there is an exact space between the main line and the location reference at the end of the line. E.g.,
- `%arg0: !tt.ptr<f32> {tt.divisibility = 16 : i32} loc("<source>":7:0),`
- `tt.return loc(#loc2)`

When I was writing the test, I found the extra space which annoys me (especially for the space before comma in the argument list). 

I have no preference between matching ` ` or `\s` (I suppose that's what you meant), but since you mentioned I will change it to `\s` to be consistent with other regex.